spark/src/main/java/org/apache/iceberg/spark/source/IcebergBatchScan.java

```diff
@@ -115,61 +116,83 @@
   private StructType type = null; // cached because Spark accesses it multiple times
   private List<CombinedScanTask> tasks = null; // lazy cache of tasks
 
-  Reader(Table table, boolean caseSensitive, DataSourceOptions options) {
+  public IcebergBatchScan(Table table, Boolean caseSensitive, CaseInsensitiveStringMap options) {
```

Comment:
Also here, why the constructor changes to `public`? Besides, `Boolean` should be changed to primitive type.